packages/panic/index.js

```diff
@@ -0,0 +1,59 @@
+/* global globalThis */
+
+// Modeled on `PassStyleOfEndowmentSymbol` of `@endo/pass-style`.
+export const PanicEndowmentSymbol = Symbol.for('@endo panic');
+
+/**
+ * Last resort fallback that violates the `panic` spec by throwing rather
+ * than immediately exiting, throws this `lastResortError`.
+ * To help prospective catchers to,
+ * reliably-enough, distinguish this caught error from other errors,
+ * we add a property
+ * named by the `PanicEndowmentSymbol` registered symbol. We use a
+ * registered symbol rather than a novel subclass of Error to avoid
+ * [Eval Twin Problems](https://github.com/endojs/endo/issues/1583).
+ * However, as a necessary price for avoiding Eval Twim Problems, this
+ * marking is forgeable -- anyone can create and throw a similar error.
+ *
+ * We also export this error so that importers can use it as an identity check.
+ * This is not forgeable, i.e., not give false positives, but due to the
+ * Eval Twin Problem, may produce falso negatives. Use this identity check
+ * with caution.
+ */
+export const lastResortError = ReferenceError('Should have already exited');
+lastResortError[PanicEndowmentSymbol] = 'Should have already exited';
+Object.freeze(lastResortError);
+
+/**
+ * Ponyfill the `panic` function of the
+ * [Don't Remember Panicking](https://github.com/tc39/proposal-oom-fails-fast)
+ * tc39 proposal.
+ *
+ * @param {Error} [err]
+ * @returns {never}
+ */
+export const panic = (err = RangeError('Panic')) => {
+  if (globalThis.console && typeof globalThis.console.error === 'function') {
+    globalThis.console.error('Panic', err);
+  } else {
+    // TODO use Moddable XS print function if we can reliably distinguish it
+    // from `print` in browsers.
+  }
+
+  if (typeof globalThis[PanicEndowmentSymbol] === 'function') {
+    /** @type {never} */ (globalThis[PanicEndowmentSymbol](err));
+  } else if (
+    globalThis.process &&
+    typeof globalThis.process.abort === 'function'
+  ) {
+    globalThis.process.abort();
+  } else {
+    // TODO Once Moddable provides a host `panic` function, use that here.
+  }
+  throw lastResortError;
```

Comment:
Or `for(;;)` in some environments?

Reply:
From the README:

> (As noted in the proposal, a higher fidelity emulation could, as a last resort, go into an infinite loop. But the consequences of this are too painful for both manual and CI testing. Besides, on some engines (browsers), in violation of the current JS spec, resume execution of user-code within the agent after the "infinite" loop exceeds a timeout. So even this strategy would not be safe on such engines.)

The last resort is what happens if the user has forgone the first two opportunities to advise the ponyfill. If the user does neither, how should the ponyfill decide between throwing vs an infinite loop?

Reply:
Resolving with this unchanged. If there is another test the ponyfill could do to decide, it could get added in a later PR.